project_development_tips.md

```diff
@@ -0,0 +1,119 @@
+- If there isn’t already an open source repo, the owner should create a public GitHub repo from [the open source template](https://github.com/onflow/open-source-template) and make sure all of this is set up with some initial documentation for what the repo is for before any code is written.
+- The owner should create a design document that lays out the intended high level design and architecture of the smart contract and some basic user stories, as well as any questions that still need to be answered about it.
+    - Diagrams should be made where applicable describing state machines, user flows, etc
+    - This document should be shared in an issue in the open source repo where the contracts or features are being developed.
+- If the owner has to move to another team, a new owner on the project’s team should be identified and ramped up immediately so that there is always an owner who is intimately involved with the project and is aware of all the details.
+- Keys for the contract and/or admin accounts should be stored in a secure key store such as google KMS or something similar, and the owner should have access to these along with the secondary owner and some other engineering manager. Keys should never be in any file in the public repo.
+- Project owners need to always keep [composability and extensibility](https://www.notion.so/Smart-Contract-Composability-be544f21a2e84596affad73952b21d56) in mind while designing, developing, and documenting their projects.
+- Cadence best practices need to be kept in mind:
+    - [https://developers.flow.com/cadence/design-pattern](https://developers.flow.com/cadence/design-patterns)s
+    - [https://developers.flow.com/cadence/anti-patterns](https://developers.flow.com/cadence/anti-patterns)
+    - [https://developers.flow.com/cadence/security-best-practices](https://developers.flow.com/cadence/security-best-practices)
+
+## **Development Process**
+
+- The same person who writes the code must also write the tests.
+    - They have the clearest understanding of the code paths and edge cases.
+- Development process should be iterative, if possible.
+    - Develop MVP first, get reviews, and test thoroughly, then add additional features with tests.
+    - Makes the review process easier.
+- Comments and field/function descriptions need to be written at the same time (or even before) the code is written
+    - Helps the developer and reviewers understand the work-in-progress code better, as well as the intentions of the design (for testing and reviewing)
+- Functions should be commented with a
+    - Description
+    - Parameter descriptions (unless it is obvious)
```

Comment:
I like to comment parameters and returns even if they are obvious so the cadence doc tool makes it pretty when we actually use to include auto generated docs on the projects.

Reply:
thats a good point. I'll remove the part in parentheses